Create SQLite example

kangalio opened the pull request:
This PR adds an example which shows using SQLx and SQLite to write a simple todo bot.

I added a README.md because using SQLx effectively requires a bit of background knowledge.

kangalio committed:
Create SQLite example

vicky5124 commented:
Some small stuff need to be worked on before merging this: (1-2 break CI/CD)

1. Missing `offline` feature on `sqlx`.
2. Missing `sqlx-data.json` generated from `cargo sqlx prepare`
3. Example number 15 merge conflicts with #1476  and should be 16 to be safe
4. The example was not added to `/Makefile.toml` nor `/examples/README.md`
5. The Actions example check name is 14
6. It should probably be mentioned that `query!` automatically generates its own structs with the fields and types of the database, and that `query_as!` should be used if a custom structure is preferred.

kangalio committed:
Fix all problems vicky found

- Added `offline` feature to sqlx dependency
- Commited `sqlx-data.json`
- Switched example number from 15 to 16
- Added example to /Makefile.toml and /examples/README.md
- Fixed copy-paste error in GitHub actions example number
- Added a small section in the example README.md which says that for detailed SQLx usage info (like `query!` vs `query_as!`) you can check their README

kangalio committed:
Fix small README formatting error

kangalio commented:
The Makefiles for the new example may not work I think because they don't supply DATABASE_URL. I don't know enough about that Makefile-system to check or fix that though

vicky5124 commented:
sqlx-cli reads `.env` so it's not needed, and the readme already says to use `.env` so it should work.

kangalio committed:
Merge branch 'current' into sqlite-example
